dsl/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+extern crate quote;
+extern crate litrs;
+extern crate proc_macro2;
+
+use quote::quote;
+
+use proc_macro2::TokenTree;
+use proc_macro2::TokenStream;
+use proc_macro2::TokenTree::Punct;
+use proc_macro2::TokenTree::Ident;
+use proc_macro2::token_stream::IntoIter;
+use std::iter::Peekable;
+
+use std::convert::TryFrom;
+use std::str::FromStr;
+use litrs::Literal;
+
+#[derive(PartialEq)]
+enum ExprType {
+    Nil, Int, Bool
+}
+
+// within the compiler, these generate tokens for
+// Value
+struct ParsedExpr {
+    t: ExprType,
+    tokens: TokenStream,
+}
+
+fn parse_literal(it: &mut Peekable<IntoIter>) -> ParsedExpr {
+    let token = it.next().expect("failed to parse literal, hit end of string");
+
+    match token {
+        Punct(ref x) => if x.as_char() == '(' {
+            let v = parse_expr(it);
+
+            it.next().expect("expected closing paren, hit end of string");
+            return v
+        },
+
+        // boolean literals
+        Ident(ref x) => match x.to_string().as_str() {
+            "true"  => return ParsedExpr{ t: ExprType::Bool, tokens: quote!{ true }},
```

Comment:
You should probably re-use the tokens you parsed instead of making a new `TokenStream` with `quote!`. This way you would be able to keep span information

Reply:
Another more abstract reason is that you won't be repeating `true` and `true` and making the compiler do extra work.

Reply:
Created #3787